packages/shared/src/hooks/usePostById.ts

```diff
@@ -38,36 +37,21 @@ export const updatePostCache = (
     },
   }));
 
-const usePostById = ({
-  id,
-  isFetchingNextPage,
-  options = {},
-}: UsePostByIdProps): UsePostById => {
+const usePostById = ({ id, options = {} }: UsePostByIdProps): UsePostById => {
   const { tokenRefreshed } = useAuthContext();
   const key = getPostByIdKey(id);
-  const {
-    data: postById,
-    isLoading,
-    isFetched,
-  } = useQuery<PostData>(
+  const query = useQuery<PostData>(
     key,
     () => request(graphqlUrl, POST_BY_ID_QUERY, { id }),
     {
       ...options,
       enabled: !!id && tokenRefreshed,
     },
   );
-
+  const { data: postById } = query;
   const post = postById || (options?.initialData as PostData);
 
-  return useMemo(
-    () => ({
-      post: post?.post,
-      isFetched,
-      isLoading: isLoading || isFetchingNextPage,
-    }),
-    [id, postById, isLoading, options, isFetched, isFetchingNextPage],
-  );
+  return useMemo(() => ({ post: post?.post, query }), [query]);
```

Comment:
Missing `post?.post` in dependency array.

Reply:
I mean, we removed redundant dependencies in the past, not sure we would want this one now. But I'm okay with it. Now I'm confused about when we want redundant dependencies and when we do not.

Reply:
> Now I'm confused about when we want redundant dependencies and when we do not.

We simply need to put all variables that are referenced inside the memo callback inside dependency array. `exhaustive-deps` rule would help with this since it would automatically say which ones are missing.

Reply:
Right, it would probably be best to enable the linting to have a general rule. Though it would be a tedious one so we will have to do it in one go for the existing implementations.

Reply:
We have a task already for the linting, but only after this work is done :)